### PR TITLE
Add sample optimized for minikube

### PR DESCRIPTION
**Pull request by nirs**
Adding sample that can be installed on ramen minikube based test environment.

The new vm should probably use an overlay so we don't duplicate the vm and pvc.
I started with copying to simplify testing.

Tested with
https://github.com/RamenDR/ramen/pull/1096

### CODE REVIEW

File: channel/channel.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+apiVersion: apps.open-cluster-management.io/v1
+kind: Channel
+metadata:
+  name: kubevirt-gitops
+spec:
+  type: GitHub
+  pathname: https://github.com/nirs/ocm-kubevirt-samples.git
```
Comment (nirs) [inline channel/channel.yaml]: Need to update to this github repo before merging.

File: subscription/subscription.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---
+apiVersion: apps.open-cluster-management.io/v1
+kind: Subscription
+metadata:
+  annotations:
+    apps.open-cluster-management.io/github-branch: ramen-test
```
Comment (nirs) [inline subscription/subscription.yaml]: Should change to main before merging

File: README.md [hunk 3]
```diff
@@ -1 +1,199 @@
...
+clusters in the default clusterset.
+
+The subscription is optimized for *Ramen* minikube based test
+environment. To use in another setup you may need to modify the
+resources.
+
+### Enabling DR for the VM
+
+To allow *Ramen* to protect the VM, you need to disable *OCM*
+scheduling by adding an annotation to the VM placement:
```
Comment (aglitke) [inline README.md:74]: Is this a bug in the OCM scheduling?  Why is it needed for VMs?

**Comment by nirs** [comment]
Converted to draft for debugging ssh key issues with kubevirt image.

**Review by aglitke — COMMENTED** [review]
looks good.  Really happy to have this documented too!  Maybe add some info about the ramen minikube env too?

File: README.md (hunk 3 above)
Comment (nirs) [reply README.md:74]: Sadly this is the official way to make ramen the scheduler for the placement :-)

**Comment by nirs** [comment]
> looks good. Really happy to have this documented too! Maybe add some info about the ramen minikube env too?

Maybe a short paragraphs and a link to https://github.com/RamenDR/ramen/tree/main/test?

File: vm-standalone-pvc-k8s/source.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+apiVersion: cdi.kubevirt.io/v1beta1
+kind: VolumeImportSource
+metadata:
+  name: cirros-source
+spec:
+  source:
+    registry:
+      # Image customized for ramen testing.
+      # TODO: Consume from ramendr repo when available.
+      url: "docker://quay.io/nirsof/cirros:0.6.2"
```
Comment (nirs) [inline vm-standalone-pvc-k8s/source.yaml]: This image includes now the ramen logging service, starting when the vm
starts and logging every 10 seconds to /var/log/ramen.log.

We can use this to track how the amount of data lost during failover and
the time when we the vm was down during relocate.

Example usage:

```
$ virtctl ssh cirros@sample-vm -n kubevirt-sample --context dr1 --command "tail -f /var/log/ramen.log"
Wed Oct 25 23:43:12 UTC 2023
Wed Oct 25 23:43:22 UTC 2023
Wed Oct 25 23:43:32 UTC 2023
Wed Oct 25 23:43:42 UTC 2023
```

**Comment by mhenriks** [comment]
I haven't tried it yet but looks great!

**Comment by nirs** [comment]
I updated the cirros image tag to 0.6.2-1. This image is optimized for faster boot time and starts the ramen logger earlier.

File: README.md [hunk 5]
```diff
@@ -1 +1,199 @@
...
+
+```sh
+for cluster in dr1 dr2; do
+    kubectl create ns kubevirt-sample --context $cluster
+    kubectl create secret generic my-public-key \
+        --from-file key1=$HOME/.ssh/id_rsa.pub \
+        --namespace kubevirt-sample \
+        --context $cluster
+done
+```
```
Comment (nirs) [inline README.md]: Testing replacement by generating the secret in the kustomization:
https://github.com/nirs/ramen/commit/dd0e3ee605bec720670f72e6a198befc7bfa361e

With this change the only manual step is:

```
cp ~/.ssh/id_rsa.pub vm-standalone-pvc-k8s/
```

So this is trivial to customize for every user, and forgetting to copy the key will
fail early with a clear message. If we create some scripts to automated test this step can be automated.

I tried another approach - adding static key to the repo, but this causes false alarms
for the security team (got mail about possible leak of the private key) so it is not
the right direction.

Another approach is to automated creating new keypair for testing, but `virtctl
--identity-file` is broken (value is ignored when using the native ssh client) so
we need to expose the vm ssh port via a service (e.g. NodePort) and use the standard
ssh client. Seems too complicated so I'm starting with the simple approach.

@aglitke what do you think?

Comment (nirs) [reply README.md]: Tested, it cannot work with OCM - the key must be in the remote git repo.
Posting a working solution soon.

**Comment by nirs** [comment]
Latest version improves ssh key management - this is an example commit for customizing this repo with your own ssh public key:
https://github.com/nirs/ocm-kubevirt-samples/commit/66babbb240d8a8295163601f201bba326e96b81c

**Comment by nirs** [comment]
Latest change: link to user quick start guide for setting up ramen test environment:
https://github.com/RamenDR/ramen/blob/main/docs/user-quick-start.md

**Pull request merged**